### PR TITLE
root: fix macos build

### PR DESCRIPTION
**Pull request by sethrj**
No ROOT `builtin` should ever be set to true, because that vendors an existing library that spack doesn't know about.

Worse, using `builtin_glew` forces the package to be on, even when not building x/gl/aqua on macos, which causes build failures. This PR fixes that problem.

Caused by https://github.com/spack/spack/pull/45632#issuecomment-2276311748 . Perhaps @stephenswat you can post the build error and we can see if there's an alternate solution that works for both cases?

### CODE REVIEW

**Review by stephenswat — APPROVED** [review]
Sorry for the inconvenience; I will be able to look at the original build error after a short weekend holiday, so I'd suggest we merge this and then I will get back to you on this next week.

**Comment by sethrj** [comment]
Could one of the ROOT maintainers (@vvolkl ) please approve when you have a chance? Thanks!

**Review by gartung — APPROVED** [review]

**Comment by sethrj** [comment]
Thanks @gartung !

**Pull request merged**
